pkg/operator/dedicatedetcdcontroller/dedicated_etcd_controller.go

```diff
@@ -0,0 +1,217 @@
+	etcdURLHost, err := dnshelpers.GetEscapedPreferredInternalIPAddressForNodeName(network, destinedEventNode)
+	if err != nil {
+		return fmt.Errorf("failed to get etcd URL hostname for node %s: %w", destinedEventNode.Name, err)
+	}
+
+	depClient := c.kubeClient.AppsV1().Deployments(operatorclient.TargetNamespace)
+
+	renderedDeployment, err := ceohelpers.RenderTemplate("etcd/dedicated-event-etcd-deployment.yaml",
+		&EventsEtcdDeploymentSubstitution{Image: c.etcdImage, NodeName: destinedEventNodeHostname})
+	if err != nil {
+		return fmt.Errorf("failed to render events-etcd deployment: %w", err)
+	}
+
+	var deployment appsv1.Deployment
+	if err := yaml.Unmarshal([]byte(renderedDeployment), &deployment); err != nil {
+		return fmt.Errorf("failed to unmarshal events-etcd deployment: %w", err)
+	}
+
+	svcTemplate := bindata.MustAsset("etcd/dedicated-event-etcd-svc.yaml")
+	var svc corev1.Service
+	if err := yaml.Unmarshal(svcTemplate, &svc); err != nil {
+		return fmt.Errorf("failed to unmarshal events-etcd svc: %w", err)
+	}
+
+	if _, err = depClient.Get(ctx, deployment.Name, metav1.GetOptions{}); err != nil {
+		if errors.IsNotFound(err) {
+			_, err := depClient.Create(ctx, &deployment, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to create events-etcd deployment: %w", err)
+			}
+		} else {
+			return fmt.Errorf("failed to get events-etcd deployment: %w", err)
+		}
+	}
+
+	_, err = depClient.Update(ctx, &deployment, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update events-etcd deployment: %w", err)
+	}
+
+	svcClient := c.kubeClient.CoreV1().Services(operatorclient.TargetNamespace)
+	if _, err := svcClient.Get(ctx, svc.Name, metav1.GetOptions{}); err != nil {
+		if errors.IsNotFound(err) {
+			_, err := svcClient.Create(ctx, &svc, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to create events-etcd service: %w", err)
+			}
+		} else {
+			return fmt.Errorf("failed to get events-etcd service: %v", err)
+		}
+	}
+
+	updatedSvc, err := svcClient.Update(ctx, &svc, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update events-etcd service: %w", err)
+	}
+
+	kas, err := c.operatorClient.KubeAPIServers().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get kas cluster: %w", err)
+	}
+
+	existingUnsupportedConfigOverrides := map[string]interface{}{}
+	if err := json.NewDecoder(bytes.NewBuffer(kas.Spec.UnsupportedConfigOverrides.Raw)).Decode(&existingUnsupportedConfigOverrides); err != nil {
+		return fmt.Errorf("failed to decode existing unsupported config overrides: %w", err)
+	}
+
+	/*
+			apiVersion: operator.openshift.io/v1
+			kind: KubeAPIServer
+			spec:
+			  unsupportedConfigOverrides:
+				apiServerArguments:
+				  etcd-servers-overrides:
+				  - "/events#https://10.0.0.5:20379"
+		  --- OR via SVC, as done below ---
+			 - "/events#https://events-etcd.openshift-etcd.svc:20379"
```

Comment:
this requires `dnsPolicy=ClusterFirstWithHostNet` on the kube-apiserver static pods

 https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy